tijanabekic/assignment2/Assignment 2.java

```diff
@@ -0,0 +1,126 @@
+//Q1. Given a binary tree and a key, write a function that prints all the ancestors of the key in the given binary tree.
+
+public class Node {
+	int data;
+	Node left;
+	Node right;
+	Node parent;

```

Comment:
If you set parent and data to be final variables then you ensure that a user cannot accidentally break the tree by making a cycle.
`final Node parent`
final values can only be set in the constructor.